src/App.js

```diff
@@ -1,16 +1,44 @@
 import React from 'react';
+import { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+
+const entries = chatMessages;
 
 const App = () => {
+  const [messages, setMessages] = useState(entries);
+
+  const heartCount = (messages) => {
+    return messages.reduce((total, message) => {
+      if (message.liked) total++;
+      return total;
+    },0);
+  };
+
+  const changeHeartColor = (id) => {
+    setMessages((oldmessages) => {
+      return oldmessages.map((message) => {
+        if (message.id === id) {
+          return {...message, liked: !message.liked}
+        }
+        return message;
+      });
+    });
+  };
+
+  const likedMessages = heartCount(messages);
```

Comment:
consider any variables that will change dynamically, turn them into state. React listens for state hooks, but not regular variables. Sometimes this can cause issues.
```python
const [heartCount, setHeartCount] = useState(0);
```
Refactor `heartCount` function to set state after it finishes reducing.